File: Source/Aladdin.CAPI/CAPI/IProvider.cs
namespace Aladdin.CAPI
{
	///////////////////////////////////////////////////////////////////////////
	// Провайдер объектов
	///////////////////////////////////////////////////////////////////////////
	public interface IProvider : IRefObject
	{
        // имя провайдера 
        string Name { get; } 

		// перечислить хранилища объектов
		string[] EnumerateStores(Scope scope); 
        // получить хранилище объектов
        SecurityStore OpenStore(Scope scope, string storeName);     

        // перечислить все объекты
		SecurityInfo[] EnumerateAllObjects(Scope scope); 
	}
}

File: Source/Aladdin.CAPI.Java/src/aladdin/capi/IProvider.java
package aladdin.capi;
import aladdin.*; 
import java.io.*; 

///////////////////////////////////////////////////////////////////////////
// Провайдер объектов
///////////////////////////////////////////////////////////////////////////
public interface IProvider extends IRefObject
{
    // имя провайдера 
    String name();  

	// перечислить хранилища объектов
	String[] enumerateStores(Scope scope); 
    // получить хранилище объекта
    SecurityStore openStore(Scope scope, String storeName) throws IOException;     

    // перечислить все объекты
	SecurityInfo[] enumerateAllObjects(Scope scope); 
}
